nebula_ros/src/velodyne/velodyne_ros_wrapper.cpp

```diff
@@ -0,0 +1,265 @@
+  config.data_port = declare_parameter<uint16_t>("data_port", 2368, param_read_only());
+  config.gnss_port = declare_parameter<uint16_t>("gnss_port", 2369, param_read_only());
+  config.frame_id = declare_parameter<std::string>("frame_id", "pandar", param_read_write());
+
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor = param_read_write();
+    descriptor.additional_constraints = "Angle where scans begin (degrees, [0.,360.])";
+    descriptor.floating_point_range = float_range(0, 360, 0.01);
+    config.scan_phase = declare_parameter<double>("scan_phase", 0., descriptor);
+  }
+
+  config.min_range = declare_parameter<double>("min_range", 0.3, param_read_write());
+  config.max_range = declare_parameter<double>("max_range", 300., param_read_write());
+  config.packet_mtu_size = declare_parameter<uint16_t>("packet_mtu_size", 1500, param_read_only());
+
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor = param_read_write();
+    descriptor.additional_constraints = "from 300 to 1200, in increments of 60";
+    descriptor.integer_range = int_range(300, 1200, 60);
+    config.rotation_speed = declare_parameter<uint16_t>("rotation_speed", 600, descriptor);
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor = param_read_write();
+    descriptor.integer_range = int_range(0, 360, 1);
+    config.cloud_min_angle = declare_parameter<uint16_t>("cloud_min_angle", 0, descriptor);
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor = param_read_write();
+    descriptor.integer_range = int_range(0, 360, 1);
+    config.cloud_max_angle = declare_parameter<uint16_t>("cloud_max_angle", 360, descriptor);
+  }
+
+  auto new_cfg_ptr = std::make_shared<const nebula::drivers::VelodyneSensorConfiguration>(config);
+  return ValidateAndSetConfig(new_cfg_ptr);
+}
+
+Status VelodyneRosWrapper::ValidateAndSetConfig(std::shared_ptr<const drivers::VelodyneSensorConfiguration>& new_config)
+{
+  if (new_config->sensor_model == nebula::drivers::SensorModel::UNKNOWN)
+  {
+    return Status::INVALID_SENSOR_MODEL;
+  }
+  if (new_config->return_mode == nebula::drivers::ReturnMode::UNKNOWN)
+  {
+    return Status::INVALID_ECHO_MODE;
+  }
+  if (new_config->frame_id.empty())
+  {
+    return Status::SENSOR_CONFIG_ERROR;
+  }
+
+  if (hw_interface_wrapper_)
+  {
+    hw_interface_wrapper_->OnConfigChange(new_config);
+  }
+  if (hw_monitor_wrapper_)
+  {
+    hw_monitor_wrapper_->OnConfigChange(new_config);
+  }
+  if (decoder_wrapper_)
+  {
+    decoder_wrapper_->OnConfigChange(new_config);
+  }
+
+  sensor_cfg_ptr_ = new_config;
+  return Status::OK;
+}
+
+void VelodyneRosWrapper::ReceiveScanMessageCallback(std::unique_ptr<velodyne_msgs::msg::VelodyneScan> scan_msg)
+{
+  if (hw_interface_wrapper_)
+  {
+    RCLCPP_ERROR_THROTTLE(get_logger(), *get_clock(), 1000,
+                          "Ignoring received PandarScan. Launch with launch_hw:=false to enable PandarScan replay.");
```

Comment:
PandarScan mention in velodyne

Reply:
Addressed in [877f90f](https://github.com/tier4/nebula/pull/148/commits/877f90f5f512ca192870b7fa4090a709b7376a8c)

Reply:
@mojomex 
While trivial, please remember that the one that writes the comment resolves the conversation :innocent: